src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -672,11 +692,26 @@ def iot_hub_get_stats(client, hub_name, resource_group_name=None):
     return client.iot_hub_resource.get_stats(resource_group_name, hub_name)
 
 
+def validate_authentication_type_input(endpoint_type, connection_string=None, authentication_type=None, endpoint_uri=None, entity_path=None):
+    is_keyBased = (AuthenticationType.KeyBased.value == authentication_type.lower()) or (authentication_type is None)
+    has_connection_string = (connection_string is not None)
+    if is_keyBased and not has_connection_string:
+        raise CLIError("Please provide a connection string '--connection-string/-c'")
+
+    has_endpoint_uri = (endpoint_uri is not None)
+    has_endpoint_uri_and_path = (has_endpoint_uri) and (entity_path is not None)
+    if EndpointType.AzureStorageContainer.value == endpoint_type.lower() and not has_endpoint_uri:
+        raise CLIError("Please provide an endpoint uri '--endpoint-uri'")
+    if not has_endpoint_uri_and_path:
+        raise CLIError("Please provide an endpoint uri '--endpoint-uri' and entity path '--entity-path'")
+
+
 def iot_hub_routing_endpoint_create(cmd, client, hub_name, endpoint_name, endpoint_type,
                                     endpoint_resource_group, endpoint_subscription_id,
-                                    connection_string, container_name=None, encoding=None,
+                                    connection_string=None, container_name=None, encoding=None,
```

Comment:
connection_string is not required anymore?

Reply:
Correct - if the routing-endpoint uses `identityBased` authentication, it does not use a connection string.